delta-app/src/main/java/io/cdap/delta/app/OffsetAndSequence.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.delta.app;
+
+import io.cdap.delta.api.Offset;
+
+/**
+ * An offset and sequence number.
+ */
+public class OffsetAndSequence {
+  private final Offset offset;
+  private final long sequenceNumber;
```

Comment:
One question here, why is the sequnceNumber for? Why we need to combine it with the offset?

Reply:
sequence number is a piece of information used by the target to ensure changes are applied correctly. Basically, if data with a higher sequence number already exists in the table, we assume the change was already applied in the past and can be ignored. You can see an example of how it is used at https://wiki.cask.co/display/CE/Change+Data+Capture#ChangeDataCapture-Example.2.  It needs to be stored with the offset because the event for that offset must match the same sequence number, otherwise the target logic no longer makes sense.